torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1091,8 +1091,20 @@ def init_variable(
                 and (
                     # pre_hook requires running arbitrary code at runtime, which we cannot handle at this time
                     # https://github.com/pytorch/pytorch/issues/139059
-                    # Check Config passed to autotuner in configs
-                    any(cfg.pre_hook is not None for cfg in kernel.configs)
+                    # we can't support pre_hook or post_hook in user defined triton kernels at the moment,
+                    # as they require the ability to execute code at runtime (AOTI can't support this)
+                    (
+                        hasattr(kernel, "user_defined_pre_hook")
+                        and kernel.user_defined_pre_hook is not None
```

Comment:
`is not None` is not correct here, as the default is `False` and `False is not None`. As written now, there will always be an error on Triton version >= 3.2.